### PR TITLE
First batch of protocol updates

### PR DESCRIPTION
**Pull request by darosior**
The latest version of revault_net brings a new serialization of messages, and new responses to messages sent to the Coordinator. Of course since it's a protocol update, we need to update all binaries at the same time.

Needs https://github.com/revault/revault_net/pull/57 , https://github.com/revault/coordinatord/pull/19 , https://github.com/revault/cosignerd/pull/14
Based on #133 

Fixes https://github.com/revault/revaultd/issues/130

### CODE REVIEW

**Comment by darosior** [comment]
I'd like to get the opportunity of moving all the server communication in `control` and unit test everything, but not sure i'll be able to keep up..

File: Cargo.toml [hunk 1]
```diff
@@ -25,7 +25,7 @@ path = "src/cli/main.rs"
 
 [dependencies]
 revault_tx = { version = "0.2", features = ["use-serde"] }
-revault_net = { git = "https://github.com/revault/revault_net" }
+revault_net = { git = "https://github.com/darosior/revault_net", branch = "always_respond" }
```
Comment (danielabrozzoni) [inline Cargo.toml:28]: Since it's merged you can use master now?

Comment (darosior) [reply Cargo.toml:28]: I just merged it 5 minutes ago..

Comment (danielabrozzoni) [reply Cargo.toml:28]: Ah sorry!

Comment (darosior) [reply Cargo.toml]: Done now :)

**Comment by darosior** [comment]
Rebased on #133

**Comment by darosior** [comment]
Rebased on master after #133 merge

File: src/daemon/control.rs [hunk 2]
```diff
@@ -707,43 +748,40 @@ pub fn share_unvault_signatures(
...
         );
-        transport.write(&serde_json::to_vec(&msg)?)?;
 
-        let res_msg: SignResponse = serde_json::from_slice(&transport.read()?)?;
+        let sign_res: SignResult = transport.send_req(&msg.into())?;
+        let signed_tx = if let Some(tx) = sign_res.tx {
+            tx
+        } else {
+            return Err(CommunicationError::CosigAlreadySigned);
+        };
```
Comment (danielabrozzoni) [inline src/daemon/control.rs:834]:
```
let signed_tx = sign_res.tx.ok_or(CommunicationError::CosigAlreadySigned)?;
```

Comment (darosior) [reply src/daemon/control.rs]: Thanks, taken

**Comment by danielabrozzoni** [comment]
ACK 717cf2bac6d770ea2b545d4829117e36c33e4908

**Review by danielabrozzoni — APPROVED** [review]

**Pull request merged**